AWSAuthSDK/Tests/AWSMobileClientTests/AWSMobileClientTests.swift

```diff
@@ -136,33 +136,67 @@ class AWSMobileClientTests: XCTestCase {
             return nil
         }).waitUntilFinished()
     }
-    
-    func signIn(username: String) {
+
+    func signIn(username: String, password: String? = nil, verifySignState: SignInState = .signedIn) {
+        let passwordToUse = password ?? sharedPassword
         let signInWasSuccessful = expectation(description: "signIn was successful")
-        AWSMobileClient.sharedInstance().signIn(username: username, password: sharedPassword) { (signInResult, error) in
+        AWSMobileClient.sharedInstance().signIn(username: username, password: passwordToUse) { (signInResult, error) in
             if let error = error {
                 XCTFail("User login failed: \(error.localizedDescription)")
                 return
             }
 
             if let signInResult = signInResult {
-                switch(signInResult.signInState) {
-                case .signedIn:
-                    print("signed in")
-                default:
-                    XCTFail("User was not signed in, un-expected.")
-                }
+                XCTAssertEqual(signInResult.signInState, verifySignState, "Could not verify sign in state")
```

Comment:
If we're not expecting signInResult to be `nil` in the normal flow of the test, we should just move this out and remove the `if let` block. Alternately, change it to a `guard let signInResult = signInResult` and `XCTFail("signInResult unexpectedly nil")` or similar

Reply:
Added check for signInResult